### ISSUE TITLE
[L3: SCREENREADER]: VET TEC - NVDA does not read the error if I forget to agree to the privacy policy

### ISSUE THREAD

**Issue by taylorkaren** [department-of-veterans-affairs/va-forms-system-core#253]
@1Copenut commented on [Tue Aug 20 2019](https://github.com/department-of-veterans-affairs/va.gov-team/issues/1141)

## Issue
Migrated from original issue: https://github.com/department-of-veterans-affairs/vets.gov-team/issues/17050

When I skip checking the **I have read the privacy policy** and press **Submit Application**, NVDA does not read out the error message. JAWS and VoiceOver both read it correctly. Screenshot attached below.

## Acceptance Criteria
As an NVDA + Firefox user, I want to understand that I must agree to the privacy policy before submitting my VET TEC application.

## Environment
* Win10 + Firefox
* NVDA
* https://staging.va.gov/education/apply-for-education-benefits/application/0994/review-and-submit

## Steps to Recreate
1. Log into the VET TEC form with test user
2. Answer questions and continue to Step 7: Review Application
3. Continue tabbing through the form until you get to the Submit Application button
4. Press `ENTER` or `SPACE` to submit, making sure you have not checked the required privacy policy
5. Verify the error message is never read out by NVDA

## Guidance from MDN
* https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_alert_role

## WCAG or Vendor Guidance (optional)
* [Labels or Instructions: Understanding SC 3.3.2](https://www.w3.org/TR/UNDERSTANDING-WCAG20/minimize-error-cues.html)
* [Error Identification: Understanding SC 3.3.1](https://www.w3.org/TR/UNDERSTANDING-WCAG20/minimize-error-identified.html)

---
![screen shot 2019-02-21 at 12 31 27 pm](https://user-images.githubusercontent.com/934879/53193032-983dbd80-35d5-11e9-9d5f-031cac489d10.png)

---

@k80bowman commented on [Mon Jan 24 2022](https://github.com/department-of-veterans-affairs/va.gov-team/issues/1141#issuecomment-1020128974)

For this one, we'll need to check the design system checkbox component and make sure that NVDA is reading the error message.

**Comment by taylorkaren** [comment]
Nick has reached out to David Kennedy regarding ticket #253. He is testing something related to it and will get back to Nick. Depending on how his tests go, Nick and Davide may pair tomorrow.

**Comment by micahchiang** [comment]
Standup aftertalk for 7/8 - The original issue this ticket covers relates to a custom presubmit component built by an app team, contrary to [this conversation on the original vets.gov-team ticket](https://github.com/department-of-veterans-affairs/vets.gov-team/issues/17050#issuecomment-467064419). The work for this should be owned by the app team responsible for the 0944 form.

**Comment by nickjg231** [comment]
David Kennedy investigated and tested this component and replied with the following:

> I’ve tested this using a reduced test case in this CodePen, set up how the design system would trigger the error. It works there in NVDA. So the best solution would be to implement that component, since we know it works there.
https://codepen.io/davidakennedy/pen/vYRLRJR?editors=1111

**Comment by taylorkaren** [comment]
Slack message sent to Kevin Hoffman and Matt Self:

The Forms Library Team is working on addressing accessibility issues in the current Forms Library system. We have determined that the root problem for one of these issues (VET TEC - NVDA does not read the error if I forget to agree to the privacy policy) is specific to the VET TEC form itself and not part of the Forms Library code.

David Kennedy has a fix mocked up in codepen (link to that is included in a comment on the ticket).

Could you please advise on who the best person or team is to be assigned this ticket and implement the code fix?

**Comment by taylorkaren** [comment]
@mrblanco1, @mattpointzxer0, and russell.lyons@accenturefederal.com (could not find GitHub handle for Russell): The Forms Library Team is working on addressing accessibility issues in the current Forms Library system. We have determined that the root problem for this issue is specific to the VET TEC form itself and not part of the Forms Library code.

David Kennedy has a fix mocked up in codepen (link to that is included in a comment above).

Please transfer this issue to the repo you work in so that the Forms Library Team can close out their work on this. 

Thank you!

**Comment by taylorkaren** [comment]
Also sent email:

Hi Russell, Abe, and Matt Self,
The Forms Library Team is working on addressing accessibility issues in the current Forms Library system. We have determined that the root problem for one of these issues (VET TEC - NVDA does not read the error if I forget to agree to the privacy policy) is specific to the VET TEC form itself and not part of the Forms Library code.David Kennedy has a fix mocked up in codepen (link to that is included in a comment on the ticket).Could you please transfer this issue to the repo you work in (or create a ticket in your system, if you are not working in GitHub) so that the Forms Library Team can close out their work on this?

Thank you

**Comment by taylorkaren** [comment]
Sent another email:

Hello Abe and others,

I am making one last attempt to identify the team that supports VET TEC form 0994. This is an online form on the va.gov site, not a Sharepoint site.

To recap, the Forms Library Team has been working on addressing accessibility issues in the current Forms Library system and we have determined that the root problem for one of these issues (VET TEC - NVDA does not read the error if I forget to agree to the privacy policy) is specific to the VET TEC form itself and not part of the Forms Library code.

David Kennedy has a fix mocked up in codepen (link to that is included in a comment on the ticket).

We would like to transfer this issue to the appropriate repo or ticketing system used by the team that supports this form so that the Forms Library Team can close out their work on this, and would appreciate an answer from anyone on this email chain who can help resolve this.

Thank you!
